src/puppetlabs/puppetdb/cli/import.clj

```diff
@@ -44,15 +49,15 @@
       (json/parse-string (archive/read-entry-content tar-reader) true))))
 
 
-(defn process-tar-entry
+(defn-validated process-tar-entry
   "Determine the type of an entry from the exported archive, and process it
   accordingly."
-  [^TarGzReader tar-reader ^TarArchiveEntry tar-entry host port metadata]
-  {:pre  [(instance? TarGzReader tar-reader)
-          (instance? TarArchiveEntry tar-entry)
-          (string? host)
-          (integer? port)
-          (map? metadata)]}
+  [^TarGzReader tar-reader :- (s/pred #(instance? TarGzReader %)
```

Comment:
Not sure if you knew about this schema feature, but you can give it a class here and it will "do the right thing". 
```
tar-reader :- TarGzReader
```
It also gives a slightly better error message since tar-gz-reader? isn't an actual function.